src/simd_256_u32.h

```diff
@@ -426,6 +426,13 @@ inline __uint128_t mul(__uint128_t a, __uint128_t b)
     return m256i_to_uint128(res);
 }
 
+/** Store low 128-bit part of `reg` to memory */
+inline void store_low(aint128* address, m256i reg)
```

Comment:
I think we could directly use `_mm256_castsi256_si128` instead of defining an helper:

> Casts vector of type __m256i to type __m128i. This intrinsic is only used for compilation and does not generate any instructions, thus it has zero latency. 

A cast should only keep the lower 128-bit (like casting an `uint32_t` to `uint16_t` only keep the lower 16-bit) and it has no runtime cost :)

Reply:
oh nice